pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -137,6 +137,13 @@ def compile(path: Path, srcpath: Path, pkg: Dict[str, Any], args):
     if (srcpath / ".built").is_file():
         return
 
+    pre = pkg.get("build", {}).get("pre")
+    if pre is not None:
+        site_packages_dir = srcpath / "install" / "lib" / "python3.9" / "site-packages"
+        pkgdir = path.parent.resolve()
+        env = {"SITEPACKAGES": str(site_packages_dir), "PKGDIR": str(pkgdir)}
+        subprocess.run(["bash", "-ce", pre], env=env, check=True)
```

Comment:
In the end this PR doesn't need it, I think? Could you please remove it from this PR and add as a separate PR ideally with a test and updated documentation https://pyodide.org/en/stable/development/new-packages.html#the-meta-yaml-file

Reply:
This might be redundant with `build/script`:

> The script section is required for a library package (build/library set to true). For a Python package this section is optional. If it is specified for a Python package, the script section will be run before the build system runs setup.py. This script is run by bash in the directory where the tarball was extracted.

Though it might be better to use `script` only for libraries and `pre` in other cases. Also we should probably improve documentation about what the working directory is in the various build steps.